### PR TITLE
[5.6] Fix dot notation in JSON translations

### PR DESCRIPTION
**Pull request by VRuzhentsov**
Also add test for old format translations

### CODE REVIEW

File: src/Illuminate/Translation/Translator.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ public function getFromJson($key, array $replace = [], $locale = null)
         // only one level deep so we do not need to do any fancy searching through it.
         $this->load('*', '*', $locale);
 
-        $line = $this->loaded['*']['*'][$locale][$key] ?? null;
+        $line = Arr::get($this->loaded['*']['*'][$locale], $key) ?? null;
```
Comment (jmarcher) [inline src/Illuminate/Translation/Translator.php]: If you are using `Arr::get()` there is no need for `?? null` anymore

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Comment by taylorotwell** [comment]
There is no explanation.

**Comment by VRuzhentsov** [comment]
@taylorotwell @sisve @tillkruss 
I previously used translations in the usual way, through php arrays. Dot notations works well, but at one stage of development I needed to build a js client, with prebuilded translations, and do not fetch them from the backend every time I booted client, and store it in `Local storage`.

At this point, I decided that it's better to export the php arrays to json file, but I ran into the problem that the current method of getting the value from the key `__('key')` ... `(new Translator)->getFromJson()` only works in the single root level of the json object, and if not found key in json, start to search in php translations.

My php source:
```php 
// resources/lang/en/emails.php

return [
    'hello'           => 'Hello',
    'account'         => [
        'password_changed' => [
            'text'  => 'You have succesfully changed your password.',
            'title' => 'Your password has been changed.',
        ],
    ],
];
```
Its conversion to json:
```json
// resources/lang/en.json
{
  "emails": {
    "hello": "Hello",
    "account": {
      "password_changed": {
        "text": "You have succesfully changed your password.",
        "title": "Your password has been changed."
      }
    }
  }
}
```
BUT:
If we call `__('emails.account.password_changed.text')`, we take the php version of the translations, because this key is not found in json
OK
For example. Create new:
```json
// resources/lang/en.json

{
  "emails": {
    "account": {
      "password_changed": {
        "text": "1",
      }
    },
    "account.password_changed.text": "2",
  },
  "emails.account.password_changed.text": "3",
}
```
and call `__('emails.account.password_changed.text')`, output will be `3`.
OK
removing key with value `3`:
```json
// resources/lang/en.json

{
  "emails": {
    "account": {
      "password_changed": {
        "text": "1",
      }
    },
    "account.password_changed.text": "2",
  },
}
```
and call `__('emails.account.password_changed.text')` again
output will be `Your password has been changed.`. neither 1 nor 2. Key not found in json, and fetching php value

by the way, i just used the same `Arr::get()` method from php implementation in `Translator` class from `get` method

I also wrote tests that show that getting a value from the root level works the same as before.

and I can already imagine what the .json files look like to those who now use this feature:
```json
{
 	"activity.columns.description": "Description",
 	"attachments.upload_files": "Upload files",
	"emails.account.password_changed.text": "You have succesfully changed your password.",
	"emails.account.password_changed.title": "Your password has been changed."
}
```
single level json, for me like a hell

Conclusion:
This feature was introduced in 5.4, but not covered enough with tests, and has not been completed
It's too late to fix the fix in 5.4, but we can still fix the situation, 5.5 LTS can still be edited.
I suggest not a feature, but a fix of previously introduced in the stable, but the finished feature

**Pull request closed without merging**